Aquarius/Src/Aquarius/Events/KeyboardEvent.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "Event.h"
+
```

Comment:
So I just wanted to test functionality of this, and it works but I had to add:
```C++
#include "Aquarius/Core/Input.h"
#include "Aquarius/Core/Log.h"
```
here.  But what I did to test was change using the escape key to close the window to be the application subscribing to a keyPressedEvent instead of using input polling. This is how I did that:
```C++
Application::Application(std::string&& windowName)
		: m_Window(Window::Create(800, 600, std::move(windowName)))
	{
		s_Application = this;

		Log::initLoggers();
		m_Window->Initialize();

		m_EventHandler.subscribe(eventType::KeyPressedEvent,
			[&](const Event& event)
			{
				try
				{
					const auto& keyEvent = dynamic_cast<const Aquarius::KeyPressedEvent&>(event);
					if (keyEvent.getCode() == Aquarius::Input::KeyCode::Key_escape)
					{
						glfwSetWindowShouldClose(m_Window->get(), true);
					}
				}
				catch (std::bad_cast&)
				{
					// Wrong event type, don't need to handle it
				}
			});

		AQ_CORE_INFO("Window Initialized Successfully");
	}
```
wIth a cheeky little lambda, and dynamic cast to cast the base Event reference down to a KeyPressedEvent so that I could call `getCode()`, this works to close the window as well, which is pretty sick!